src/finite_gp_projection.jl

```diff
@@ -36,6 +30,11 @@ end
 Base.length(f::FiniteGP) = length(f.x)
 
 (f::AbstractGP)(x...) = FiniteGP(f, x...)
+function (f::AbstractGP)(
+    X::AbstractMatrix, args...; obsdim::Union{Int,Nothing}=KernelFunctions.defaultobs
+)
+    return FiniteGP(f, KernelFunctions.vec_of_vecs(X; obsdim=obsdim), args...)
```

Comment:
Even better in case someone decides to implement `(f::T)(args...)` for some `T<:AbstractGP` in a different way:
```suggestion
    return f(KernelFunctions.vec_of_vecs(X; obsdim=obsdim), args...)
```